### PR TITLE
feat!: Replace IViewModelView in favor of IDispatcher.

### PR DESCRIPTION
**Pull request by jeanplevesque**
GitHub Issue: #

## Proposed Changes

 - Feature 

## BREAKING CHANGE
- Replace `IViewModelView` in favor of `IDispatcher`.
  This better reflects the functionality and better aligns with the MVVM pattern (because having ViewModels reference a View is closer to the MVP pattern).
  - `IDispatcher` is functionally equivalent to what `IViewModelView` was, except that it doesn't have the `Loaded` and `Unloaded` events (which weren't really used).
  - Implementations of `IViewModelView` and `IViewModelViewFactory` were renamed.
    - `BatchingViewModelView` --> `BatchingCoreDispatcherDispatcher`
    - `ViewModelView` --> `CoreDispatcherDispatcher`
    - `ViewModelViewFactory` --> `DispatcherFactory`

## Checklist

Please check if your PR fulfills the following requirements:

- [x] Documentation has been added/updated
- [x] Automated Unit / Integration tests for the changes have been added/updated
- [ ] Contains **NO** breaking changes
- [x] Updated the Changelog
- [ ] Associated with an issue

### CODE REVIEW

**Review by Soap-141 — APPROVED** [review]

File: src/DynamicMvvm.Abstractions/ViewModel/IDispatcher.cs [hunk 1]
```diff
@@ -1,31 +1,23 @@
...
 	{
 		/// <summary>
 		/// Gets whether or not the thread has dispatcher access.
 		/// </summary>
 		bool GetHasDispatcherAccess();
 
-		/// <summary>
-		/// Occurs when the view is loaded.
-		/// </summary>
-		event EventHandler Loaded;
```
Comment (vincentcastagna) [inline src/DynamicMvvm.Abstractions/ViewModel/IDispatcher.cs:22]: Should we consider this as a breaking changes to remove those events ?

**Review by vincentcastagna — APPROVED** [review]

File: src/DynamicMvvm.Abstractions/ViewModel/IDispatcher.cs (hunk 1 above)
Comment (vincentcastagna) [reply src/DynamicMvvm.Abstractions/ViewModel/IDispatcher.cs:22]: I know this was making the VM aware of the view ... I'm just wondering if it was used elsewhere than internally

Comment (jeanplevesque) [reply src/DynamicMvvm.Abstractions/ViewModel/IDispatcher.cs:22]: Yes, it's documented as such too :)

**Pull request merged**